src/main/java/com/google/api/codegen/configgen/mergers/ConfigMerger.java

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.configgen.mergers;
+
+import com.google.api.codegen.ConfigProto;
+import com.google.api.codegen.configgen.nodes.ConfigNode;
+import com.google.api.codegen.configgen.nodes.FieldConfigNode;
+import com.google.api.codegen.configgen.nodes.metadata.DefaultComment;
+import com.google.api.codegen.configgen.nodes.metadata.FixmeComment;
+import com.google.api.tools.framework.model.Model;
+
+/** Merges the gapic config from a Model into a ConfigNode. */
+public class ConfigMerger {
```

Comment:
Can we let Merge implements Visitor?

Reply:
The visitors do a different thing based on the type of the node. The mergers do a different thing based on the data in the node that corresponds to the IDL. Neither `NodeVisitor` nor `Visitor` is appropriate for the mergers.